File: 14.longest-common-prefix/csharp/tests/LongestCommonPrefixTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Leetcode;

namespace LeetcodeTest
{
    [TestClass]
    public class LongestCommonPrefixTest
    {
        Solution solution = new Solution();

        private void ProcessTestCase(string[] input, string expected)
        {
            string actual = solution.LongestCommonPrefix(input);
            Assert.AreEqual(expected, actual);
        }

        [TestMethod]
        public void have_longest_common_prefix()
        {
            string[] input = new string[] { "flower","flow","flight" };
            string expected = "fl";

            ProcessTestCase(input, expected);
        }

        [TestMethod]
        public void no_longest_common_prefix()
        {
            string[] input = new string[] { "dog","racecar","car" };
            string expected = "";

            ProcessTestCase(input, expected);
        }
    }
}

File: 14.longest-common-prefix/php/tests/LongestCommonPrefixTest.php
<?php

require 'vendor/autoload.php';

use PHPUnit\Framework\TestCase;
use Andrew\Leetcode\LongestCommonPrefix;

final class LongestCommonPrefixTest extends TestCase
{
	private function processTestCase(array $input, string $expected): void
	{
		$actual = LongestCommonPrefix::do($input);
		$this->assertEquals($expected, $actual);
	}

	final function test_have_longest_common_prefix(): void
	{
		$input = [ 'flower', 'flow', 'flight' ];
		$expected = 'fl';

		$this->processTestCase($input, $expected);
	}

	final function test_no_longest_common_prefix(): void
	{
		$input = [ 'dog', 'racecar', 'car' ];
		$expected = '';

		$this->processTestCase($input, $expected);
	}
}
